### DISCUSSION TITLE
HotROD command line argument value to set mySQL query delay to 100ms

### DISCUSSION THREAD

**Discussion by tringuyen-yw** [jaegertracing/jaeger#4731] [Q&A] [answered]
The [HotROD apllication command flags](https://github.com/jaegertracing/jaeger/blob/main/examples/hotrod/cmd/flags.go) `--help` shows:

```
Usage:
  examples-hotrod all [flags]

Flags:
  -h, --help   help for all

Global Flags:
  -b, --basepath string                  Basepath for frontend service(default "/")
  -c, --customer-service-port int        Port for customer service (default 8081)
  -d, --driver-service-port int          Port for driver service (default 8082)
  -D, --fix-db-query-delay duration      Average latency of MySQL DB query (default 300ms)
  -M, --fix-disable-db-conn-mutex        Disables the mutex guarding db connection
  -W, --fix-route-worker-pool-size int   Default worker pool size (default 3)
  -f, --frontend-service-port int        Port for frontend service (default 8080)
  -j, --jaeger-ui string                 Address of Jaeger UI to create [find trace] links (default "http://localhost:16686")
  -m, --metrics string                   Metrics backend (expvar|prometheus) (default "expvar")
  -x, --otel-exporter string             OpenTelemetry exporter (otlp|stdout) (default "otlp")
  -r, --route-service-port int           Port for routing service (default 8083)
  -v, --verbose                          Enables debug logging
```

**QUESTION** I would like to remove the mySQL lock and reduce the DB duration from default `300ms` to `100ms`. Is it correct to pass the following command line arguments? Quite unclear how to express a typed Duration value in millisecs through command line.

```
  --fix-disable-db-conn-mutex  --fix-db-query-delay=100ms
```

**Comment by yurishkuro** [comment]
`--fix-db-query-delay=100ms` is in milliseconds. Not sure I understand what the confusion is.

**Reply by tringuyen-yw** [reply]
Let's take an example of  `--fix-route-worker-pool-size` which is an `int`. To set a poolsize of 99, I am pretty confident that `--fix-route-worker-pool-size=99` would work. But the `--fix-db-query-delay` is a `duration` as hinted by the help output. 

What is the string to pass to `--fix-db-query-delay` to mean 100 millisecs? is it `100` , `100ms`, `100 ms` (with a space), `100 * time.Millisecond` . Ultimately that value is just a string. What is the string value the HotROD expects to parse, without error, to make it a duration in millisecs. Maybe it is defaulted to understand millisecs? so would it be OK to just pass it `100` ?

**Reply by yurishkuro** [reply, marked as answer]
>   `-D, --fix-db-query-delay duration      Average latency of MySQL DB query (default 300ms)`

Here the default value `300ms` is the format you provide to the flag.
